hologram_dashboard/fetcher/hologram_data_source_availability_fetcher.cc

```diff
@@ -34,4 +34,53 @@ void HologramDataSourceAvailabilityFetcher::
                                                 &hologram_configs_));
 }
 
+void HologramDataSourceAvailabilityFetcher::UpdateProto(std::string system, 
+    std::time_t time, DataSource data_source, StatusType status) {
+    std::unordered_map<DataSource,HologramDataAvailability>& 
+        data_to_availability_map = data_sources_availability_map_[system];
+    HologramDataAvailability& hda = data_to_availability_map[data_source];
+    DataSourceDetail* latest_status = hda.mutable_latest_status();
+    if (!hda.has_data_source()) {
+        hda.set_data_source(data_source);
+    }
+    latest_status->set_date((int) time);
+    latest_status->set_status(status);
+    UpdateHistory(hda, time, status);
+}
+
+void HologramDataSourceAvailabilityFetcher::UpdateHistory(
+    HologramDataAvailability& hda, std::time_t time, StatusType status) {
```

Comment:
If you are going to change content for hda, use

HologramDataAvailability* hda

Reply:
Doesn't pass by reference achieve the same result?

Reply:
updated to pointer either way